noncer.go

```diff
@@ -0,0 +1,68 @@
+package oauth1
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+)
+
+// Noncer provides random nonce strings.
+type Noncer interface {
+	Nonce() string
+}
+
+// NoncerFunc is an adapter to allow the use of
+// ordinary functions as Noncers. If f is a function
+// with the appropriate signature, NoncerFunc(f) is a
+// Noncer that calls f.
+type NoncerFunc func() string
+
+// Nonce calls f().
+func (f NoncerFunc) Nonce() string {
+	return f()
+}
+
+var (
+	// DefaultNoncer is the default Noncer. It reads 32
+	// bytes from crypto/rand and returns those bytes as a
+	// base64 encoded string.
+	DefaultNoncer Noncer = Base64Noncer{
+		Length: 32,
+	}
+)
+
+// Base64Noncer reads Length bytes from crypto/rand and
+// returns those bytes as a base64 encoded string. If
+// Length is 0, 32 bytes are read.
+type Base64Noncer struct {
+	Length int
+}
+
+// Nonce provides a random nonce string.
+func (n Base64Noncer) Nonce() string {
+	length := n.Length
+	if length == 0 {
+		length = 32
+	}
+	b := make([]byte, length)
+	rand.Read(b)
+	return base64.StdEncoding.EncodeToString(b)
+}
+
+// HexNoncer reads Length bytes from crypto/rand and
+// returns those bytes as a base64 encoded string. If
+// Length is 0, 32 bytes are read.
+type HexNoncer struct {
```

Comment:
Does this need to be here? Those users who have special nonce needs can implement their own. Not sure hex is special or just your preferred variant.

Reply:
I suppose, go ahead and keep it. If we're going to export an interface, there should be two concrete types that implement it.

Reply:
Doesn't need to be there, and I'm good either way. I know what I need to use and I tossed it in here as an alternative option. I'll keep it in per your second comment, but I'm fine either way.